tests/testthat/test-plot_with_settings.R

```diff
@@ -58,3 +58,468 @@ testthat::test_that("if teal.plot_dpi is not an integer value then get_plot_dpi
     "teal.plot_dpi"
   )
 })
+
+
+testthat::test_that("print_plot is able to plot different types of graphics", {
+  plot_funs <- list(
+    function() {
+      print(ggplot2::ggplot(mtcars, ggplot2::aes(mpg, wt)) +
+        ggplot2::geom_point())
+    },
+    function() lattice::densityplot(1),
+    function() ggplotify::as.grob(lattice::densityplot(1)),
+    function() plot(1),
+    function() boxplot(2),
+    function() 2
+  )
+
+  plot_types <- list(
+    function() "ANYTHING",
+    function() "ANYTHING",
+    function() "grob",
+    function() "base",
+    function() "base",
+    function() "other"
+  )
+
+  for (p in seq_along(plot_funs)) {
+    testthat::expect_true(
+      is_draw(function() print_plot(plot_funs[[p]], plot_types[[p]]))
+    )
+  }
+})
+
+data <- structure(
```

Comment:
to consider: moving everything that is not a testthat to helpers

Reply:
helpers are not for most of the data in my opinion. We want to have the test and data bind, for easy access and clear link.